### PR TITLE
MGMT-18905: Fix failing dependencies

### PR DESCRIPTION
**Pull request by eifrach**
- Fix for depndabot failing to update packages.

- Fix issue when sometime the URL  `https://dl.k8s.io/release/stable.txt`  return's 500 when gettting the latest version for kubectl
- Chaging code to make it a bit more readble and simple  

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by adriengentil — APPROVED** [review]

**Review by adriengentil — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -81,13 +81,13 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
```
Comment (paul-maidment) [inline go.mod:81]: As I understand it, these changes are being performed to correctly apply the following version bump attempt by dependabot

https://github.com/openshift/assisted-service/pull/6747/files

The changes here make sense, I can see that you have bumped the package versions to the highest possible versions while still remaining on the same go version.

I also see some general tidy up such as the addition of the `-s` (silent) flag to curl operations and the formal download of the correct oc version for the architecture. All of these are sensible changes but I think we should mention that we are doing both of these things here.

**Review by paul-maidment — CHANGES_REQUESTED** [review]
Only really changes to the description of the PR I think

Mention what dependencies you are tryint to fix and what dependabot job failed, maybe a link to the job.

Mention that you are also doing some general code cleanup and summarise it.

File: go.mod (hunk 1 above)
Comment (eifrach) [reply go.mod:81]: Roger, 
This reason I changed the client download was that some time the url `https://dl.k8s.io/release/stable.txt` was returning code  500 sometimes. 

+  it's not containing the `oc` tool which is include in the other link

**Comment by danmanor** [comment]
@eifrach Did you tested it ? I mean cloning a failing `dependabot` PR and see that it passes the tests with these changes

**Comment by eifrach** [comment]
> @eifrach Did you tested it ? I mean cloning a failing `dependabot` PR and see that it passes the tests with these changes

yes. 
I tested it on my repo.  because we don't have permission for this repo it's hard to debug it

**Pull request merged**